tests/generation/test_utils.py

```diff
@@ -194,6 +194,7 @@ def _greedy_generate(
         output_attentions=False,
         output_hidden_states=False,
         return_dict_in_generate=False,
+        use_cache=True,
```

Comment:
changes in this file: pass `use_cache` to `generate`, instead of relying on `model.config.use_cache=False` and its side-effects

added a check to confirm that the cache is None when we pass `use_cache=False`